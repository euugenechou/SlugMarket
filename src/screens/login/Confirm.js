// React imports 
import React from "react";
import {
  Text,
  TextInput,
  StyleSheet,
  View,
  Keyboard,
  TouchableWithoutFeedback
} from "react-native";
import Icon from "react-native-vector-icons/Entypo";
import { Button } from "react-native-elements";
// AWS imports 
import { Auth } from "aws-amplify";
// Local imports
import TopAlert from "../components/TopAlert";

export default class App extends React.Component {
  state = {
    confirmationCode: "",
    visible: false
  };

  static navigationOptions = () => ({
    headerTintColor: "teal",
    headerStyle: {
      height: 40,
      backgroundColor: "white",
      borderBottomWidth: 0
    }
  });

  onChangeText(key, value) {
    this.setState({
      [key]: value
    });
  }

  confirmSignUp() {
    Auth.confirmSignUp(
      this.props.navigation.getParam("userName"),
      this.state.confirmationCode
    )
      .then(() => this.props.navigation.navigate("SignInScreen"))
      .catch(() =>
        this.setState({
          visible: true,
          errorMessage: "Confirmation code incorrect"
        })
      );
  }

  render() {
    return (
      <TouchableWithoutFeedback onPress = { Keyboard.dismiss } accessible = { false }>
        <View style = { styles.container }>
          <TopAlert
            visible = { this.state.visible }
            message = { this.state.errorMessage }
          />
          <Icon
            name = "chevron-small-left"
            size = { 45 }
            color = "teal"
            style = { {paddingVertical: 30} }
            onPress = { () => this.props.navigation.navigate("SignUpScreen") }
          />
          <View style = { styles.innerContainer }>
            <Text style = { styles.text }> Confirmation Code </Text>
            <TextInput
              onChangeText = { value => this.onChangeText("confirmationCode", value) }
              style = { styles.input }
              keyboardType = "number-pad"
              placeholder = "e.g.: 123456"
              placeholderTextColor = "gray"
              autoCapitalize = "none"
              autoCorrect = {false}
            />
            <Button
              raised
              color = "white"
              title = "Confirm Sign Up"
              fontWeight = "bold"
              backgroundColor = "teal"
              borderRadius = { 5 }
              containerViewStyle = { styles.buttonContainer }
              onPress={ () => this.confirmSignUp() }
            />
          </View>
        </View>
      </TouchableWithoutFeedback>
    );
  }
}

const styles = StyleSheet.create({
  input: {
    alignContent: "center",
    alignSelf: "center",
    height: 50,
    color: "black",
    borderBottomWidth: 0.7,
    borderBottomColor: "darkgray",
    marginBottom: 20,
    width: 300
  },
  container: {
    flex: 1,
    backgroundColor: "white"
  },
  innerContainer: {
    flex: 1,
    backgroundColor: "white",
    justifyContent: "center",
    paddingBottom: 90,
  },
  buttonContainer: {
    width: 300,
    paddingTop: 10,
    alignSelf: "center"
  },
  text: {
    paddingLeft: 32,
    fontSize: 20,
    color: "black",
    fontWeight: "700",
    textAlign: "left"
  }
});
