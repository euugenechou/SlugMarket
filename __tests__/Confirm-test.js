// __tests__/Confirm-test.js
import 'react-native';
import React from 'react';
import Confirm from '../src/screens/login/Confirm';

import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <Confirm />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});