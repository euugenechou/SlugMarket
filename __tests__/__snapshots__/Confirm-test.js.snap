// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={false}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Array [
        Object {
          "color": "teal",
          "fontSize": 45,
        },
        Object {
          "paddingVertical": 30,
        },
        Object {
          "fontFamily": "Entypo",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    ï……
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "flex": 1,
        "justifyContent": "center",
        "paddingBottom": 90,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontSize": 20,
          "fontWeight": "700",
          "paddingLeft": 32,
          "textAlign": "left",
        }
      }
    >
       Confirmation Code 
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      autoCorrect={false}
      keyboardType="number-pad"
      onChangeText={[Function]}
      placeholder="e.g.: 123456"
      placeholderTextColor="gray"
      style={
        Object {
          "alignContent": "center",
          "alignSelf": "center",
          "borderBottomColor": "darkgray",
          "borderBottomWidth": 0.7,
          "color": "black",
          "height": 50,
          "marginBottom": 20,
          "width": 300,
        }
      }
      underlineColorAndroid="transparent"
    />
    <View
      style={
        Array [
          Object {
            "marginLeft": 15,
            "marginRight": 15,
          },
          undefined,
          Object {
            "alignSelf": "center",
            "paddingTop": 10,
            "width": 300,
          },
          Object {
            "borderRadius": 5,
          },
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={null}
      >
        <View
          pointerEvents="box-only"
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#9E9E9E",
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 19,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "backgroundColor": "teal",
              },
              Object {
                "borderRadius": 5,
              },
              Object {
                "padding": 12,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 20,
                  },
                  Object {
                    "color": "white",
                  },
                  Object {
                    "fontSize": 17.5,
                  },
                  undefined,
                  undefined,
                  Object {
                    "fontWeight": "bold",
                  },
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            Confirm Sign Up
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
